### ISSUE TITLE
cquery --output=files returns _both_ src _and_ output file for py_binary

### ISSUE THREAD

**Issue by illicitonion** [bazelbuild/bazel#17703]
### Description of the bug:

For this BUILD file:
```starlark
py_binary(
    name = "pyb",
    srcs = ["pyb.py"],
)
```

This query produces _both_ the source file _and_ the output file:
```console
% bazel cquery --output=files :pyb 2>/dev/null
pyb.py
bazel-out/darwin-fastbuild/bin/pyb
```

In this case, it feels like just the generated file should be output, not the source file?

(This was the behaviour in Bazel 5, and I'm aware it was changed at my request in https://github.com/bazelbuild/bazel/pull/16602)

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```console
% touch WORKSPACE
% echo 6.1.0 > .bazelversion
% cat <<EOF >BUILD.bazel
py_binary(
    name = "pyb",
    srcs = ["pyb.py"],
)
EOF
% cat <<EOF >pyb.py
print("Hello")
EOF
% bazel cquery --output=files :pyb 2>/dev/null
pyb.py
bazel-out/darwin-fastbuild/bin/pyb
```

Expect that last command to instead act like this:
```console
% bazel cquery --output=files :pyb 2>/dev/null
bazel-out/darwin-fastbuild/bin/pyb
```

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

release 6.1.0

**Comment by rickeylev** [comment]
Yeah, it's a undesirable quirk. It's been there a long time. I was never able to track down when it came about or why; my best guess is it is some holdover from a much earlier phase in Bazel's life when what a rule produced wasn't as well defined or controllable.
